STN-351: QA Feedback 16

heatherdesigns opened the pull request:
# [STN-372](https://sparkbox.atlassian.net/browse/STN-351) READY FOR REVIEW

## Summary
Resolve issues discovered during the Sprint 16 QA Feedback session listed in STN-351

## Need Review By (Date)
5/26

## Urgency
medium

## Steps to Test
1. In the CLI run `npm run test` and confirm all of our Sass tests pass.
<br>

2. Image Styles on node pages
- [x] In the top section of the node detail page, images should maintain their size and aspect-ratio as defined by the image style. The image size should remain fixed and the text region should be flexible. Pages to review:
* http://economics.suhumsci.loc/events/qa-events-all-fields-filled
* http://economics.suhumsci.loc/news/qa-joeys-test-news-item
* http://economics.suhumsci.loc/people/qa-person-all-fields-filled
* http://economics.suhumsci.loc/events/gsb-organizational-behavior-seminars/gsb-organizational-behavior-seminars-13
<br>

3. Pills 
Find a page that is currently using pills or update a page such as http://economics.suhumsci.loc/people/faculty. 
* Update the view so that `Research Areas` Multiple Field Settings > Display Type uses an `ol` and in Style Settings > Customize field HTML add the class of `hb-pill-link-list`. 
Improve the color contrast of pills in Colorful
* Update `Fields of Interest` Multiple Field Settings > Display Type to use an `ul` and in Style Settings > Customize field HTML add the class of `hb-pill-list`.
- [x] Confirm that unordered lists do not display bullets before the pill. See screenshot in [STN-351](https://sparkbox.atlassian.net/browse/STN-351) for clarification.

- [x] Improve the color contrast of pills in Colorful
Review the colors of the pills in each color pairing (Ocean, Mountain, and Cardinal). Confirm the color contrast for each has improved and that they meet [WCAG AA](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html) guidelines.
<br>

4. More link
- [x] Review more link when located in the local footer dark variant. Confirm color contrast improvement and that it meets WCAG AA guidelines.
<br>

5. Row Background Color
Create a page filled for testing content within rows that have full width and limited width wells. The the screenshot for `row background color` in JIRA STN-351 as an example. Create the following scenarios for full width and limited width wells:
- [x] Views without a title --> look for adequate spacing on top
- [x] horizontal card postcard --> look for adequate spacing on top
- [x] accordions --> Make sure they span the width of the container and the plus sign does not overlap the text. I also added a background-color to the accordion summary as it was the background-color of the well.
- [x] hero --> Confirm they span the width of the container. I also made spacing adjustments for consistency.
- [x] carousel --> I made spacing adjustments for consistency and changed the background color of the grey block behind the image because it was the same color as the well. 
<br>

6. Mixing bullets and numbers in WYSIWYG results
- [x] Go to https://economics-prod.stanford.edu/graduate/phd-degree and confirm that the bullets and numbers within the `General Timeline` lists do not overlap upon each other.
<br>

7. Date Stacked Horizontal Card
- [x] Go to http://economics.suhumsci.loc/seminars-workshops/upcoming-seminars-and-events and confirm that the date, location, and speaker icon display for highlighted events and upcoming seminars.
<br>

8. Mailto links on postcards render the envelope icon
- [x] This was fixed in QA Feedback Sprint 15 work. Confirm fix by going to https://economics-dev.stanford.edu/undergraduate/honors-program. Scroll towards the bottom of the page and look for the `How to Apply` postcard.
<br>

9. More Link Traditional Arrow not quite centered (it is a tad too high)
- [x] On the Traditional theme, confirm the alignment of the arrows on vertical linked cards and the more link has improved.
<br>

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)

Commit:
feat: update background-color of colorful themed pills to improve color contrast / accessibility

Commit:
feat: prevent images on node details pages from shrinking

Commit:
feat: improve color contrast / accessibility of more links for the local footer dark variant

Commit:
feat: add margin top to cards within row wells when a heading is not present

Commit:
feat: add styles to allow accordions to span the width of rows within full and limited width wells

Commit:
feat: adjust spacing around heros and carousels when placed in a row with a well

Commit:
feat: adjust styles to center more link traditional arrow

Commit:
feat: write styles to prevent overlapping numbers and bullets in nested list elements

Commit:
feat: implement solution for icons created for the date stacked horizontal card to render as expected all the time

- We are using classes created by Drupal for the styling of the icons. Sometimes these classes auto-generate a modifer to the end of the class name. In an effort to cover all bases, use regex to render styles as long as the core class name matches.

heatherdesigns commented on docroot/themes/humsci/humsci_basic/src/scss/components/_card.scss:
```diff
@@ -147,14 +147,14 @@
     &--date {
       // only apply calendar icon when the label exists
-      .views-label-field-hs-event-date + .field-content {
+      [class*="views-label-field-hs-event-date"] + .field-content {
```
We are using classes generated by Drupal for the styling of the icons. Targeting `.views-label-field-hs-event-date` is fine for the highlight events but the icons were not displaying for the list of upcoming seminars. I discovered upcoming seminars has a different class name of `.views-label-field-hs-event-date-5`. I'm unsure how Drupal generates the `-5` and assume this number could be different when the date stacked horizontal pattern is used on another view. The change introduced was the only way I could think of targeting all potential scenarios.

Commit:
feat: clean up / remove comment

kaseybon commented on docroot/themes/humsci/humsci_basic/src/scss/objects/_layouts.row.scss:
```diff
@@ -7,6 +7,49 @@
+      }
+    }
+
+    // When there is only one accordion in the row, make it grow to the entire width. This property is not needed on accordions when they are placed in a row with other items.
```
👏

kaseybon commented:
There are some issues with the image overlapping the on pages with 3 columns:
http://economics.suhumsci.loc/people/christopher-makler

<img width="1285" alt="Screen Shot 2020-05-26 at 9 01 20 AM" src="https://user-images.githubusercontent.com/2486846/82903840-90cf3780-9f2f-11ea-90b6-5c9a9bcf23ff.png">

heatherdesigns commented:
## Issue discovered while browser testing:

Hero in row looks fine in Safari, but doesn't expand the full width in Chrome as some breakpoints.

![safari left vs chrome right](https://user-images.githubusercontent.com/12678977/82914661-c8dd7700-9f3d-11ea-8c63-f2f26bff649f.png)

Update: Now I see if in Safari too. 😳

Commit:
feat: add styles that allow for content in the main top text area of node details pages to break apart and flow alongside images

heatherdesigns commented:
I added the `word-break` property to allow the content beside images to collapse as needed. Unfortunately this could get ugly with large images. In my local environment I updated the image styles to used `Small Square 200 x 200` for the person node detail pages because of the 2 column column. I will share this at demo.

![faculty node](https://user-images.githubusercontent.com/12678977/82919191-b5350f00-9f43-11ea-9c43-6bd94e9df3d3.png)

Commit:
feat: ensure that hero and carousels within rows expand their full width

Commit:
feat: add compiled CSS

silverli reviewed (changes requested):
one last small thing I noted is that for an accordion with a very long title, it needs bit more padding around the arrow so they dont overlap

<img width="1213" alt="Screen Shot 2020-05-26 at 12 19 54 PM" src="https://user-images.githubusercontent.com/12848168/82926624-7c019c80-9f4d-11ea-99cc-76867c76a899.png">

Commit:
feat: add spacing to accordion's to prevent summary content from overlaying the plus icon

Commit:
feat: add compiled CSS after accordion spacing fix

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/objects/_layouts.row.scss:
```diff
@@ -27,6 +27,10 @@
           @include hb-global-color('background-color', 'white');
+
+          summary {
+            padding-right: hb-calculate-rems(52px);
```
👍

Commit:
feat: add compiled CSS

silverli reviewed (approved):
Looks great!! Awesome and thorough job on all this!
